File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    let numCount = {};
    let counter = 0;
    for(let num of nums){
      if(numCount[num]){
          counter += numCount[num];
      } 
     numCount[num] = (numCount[num]||0)+1;
    }
return counter;
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        counter = {}
        count = 0
        for num in nums:
            if num in counter:
                counter[num] += 1
            else:
                counter[num] = 1
        for key in counter:
            count += counter[key] * (counter[key] - 1) / 2

        return int(count)

        
        